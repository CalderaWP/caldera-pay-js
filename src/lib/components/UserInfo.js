// @flow
import React from 'react';
import {RenderGroup} from '@caldera-labs/components';
import type {WordPressUser} from "../types/WordPress";

type Props = {
	user: 	WordPressUser,
	onChangeEmail: Function,
	onUpdateUserFields: Function,
	onLoginAttempt: Function,
	showPassword: boolean
};
type State = {
	password: string
}
/**
 * Displays product cart
 * @param {Props} props
 * @return {*}
 * @constructor
 */
export class UserInfo extends React.Component<Props,State> {
	state = {
		password:''
	};
	/** @inheritDoc **/
	constructor(props: Props) {
		super(props);
		(this: any).setPassword = this.setPassword.bind(this);
		(this: any).handleLogin = this.handleLogin.bind(this);
	}

	/**
	 * Set password in state
	 *
	 * @param password
	 */
	setPassword(password:string){
		this.setState({password});
	}

	/**
	 * Call onLoginAttempt, passing password from state
	 */
	handleLogin(){
		this.props.onLoginAttempt(this.state.password);
	}


	/** @inheritDoc**/
	render() {

		const {props,state} = this;
		const {showPassword,user} = props;
		//@todo solve this problem in components library
		const fullWidthClassName = 'col-sm-12';
		const halfWidthClassName =  'col-sm-12 col-md-6';
		const emailClassName = showPassword ? halfWidthClassName : fullWidthClassName;



		const emailField = {
			id: 'caldera-pay-customer-email',
			label: 'Email Address',
			desc: '',
			type: 'input',
			required: true,
			innerType: 'email',
			fieldClassName: 'form-control',
			value: user.email,
			onValueChange: (newValue) => {
				props.onChangeEmail(newValue);
				props.onUpdateUserFields({
					...props.user,
					email: newValue
				});
			}
		};


		const passField = {
			id: 'caldera-pay-customer-password',
			label: 'Password',
			desc: 'Account Found Enter Password',
			type: 'input',
			required: true,
			innerType: 'password',
			fieldClassName: 'form-control',
			value: state.password,
			onValueChange: (newValue) => {
				this.setPassword(newValue);
			}
		};


		const firstNameField = {
			id: 'caldera-pay-customer-first-name',
			label: 'First Name',
			desc: '',
			required: true,
			type: 'input',
			innerType: 'text',
			fieldClassName: 'form-control',
			value: user.first_name,
			onValueChange: (newValue) => {
				props.onUpdateUserFields({
					...props.user,
					first_name: newValue
				});
			}
		};

		const lastNameField = {
			id: 'caldera-pay-customer-last-name',
			label: 'Last Name',
			desc: '',
			required: false,
			type: 'input',
			innerType: 'text',
			fieldClassName: 'form-control',
			value: user.last_name,
			onValueChange: (newValue) => {
				props.onUpdateUserFields({
					...props.user,
					last_name: newValue
				});
			}
		};



		return (
			<div
				className={'caldera-pay-customer-information'}
			>
				<div className={'row'}>
					<RenderGroup
						configFields={[emailField]}
						className={emailClassName}
					/>
					{ showPassword &&
						<RenderGroup
							configFields={[passField]}
							className={halfWidthClassName}
						/>
					}

				</div>

				<div className={'row'}>
					<RenderGroup
						configFields={[firstNameField]}
						className={halfWidthClassName}
					/>
					<RenderGroup
						configFields={[lastNameField]}
						className={halfWidthClassName}
					/>
				</div>

				{props.showPassword &&
					<div className={'row'}>
						<button
							className={'btn-green'}
							onClick={this.handleLogin}>
							Login
						</button>
					</div>

				}
			</div>
		);
	}
};
