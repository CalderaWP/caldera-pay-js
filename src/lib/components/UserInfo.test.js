import React from "react";
import renderer from "react-test-renderer";
import {UserInfo} from "./UserInfo";
import type {WordPressUser} from "../types/WordPress";

const genericHandler = () => {};
const bundleOrder = [
	'isFree',
	20520, //Individual
	20518, //Advanced
	48255, //Agency
];

describe( 'UserInfo component', () => {
	const user = {
		id: 0,
		username: '',
		name: '',
		first_name: '',
		last_name: '',
		email: ''
	}
	it('Does not show password when told not to', () => {
		const component = renderer.create(<UserInfo
				user={user}
				onChangeEmail={genericHandler}
				onUpdateUserFields={genericHandler}
				onLoginAttempt={genericHandler}
				showPassword={false}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Does show password when told  to', () => {
		const component = renderer.create(<UserInfo
				user={user}
				onChangeEmail={genericHandler}
				onUpdateUserFields={genericHandler}
				onLoginAttempt={genericHandler}
				showPassword={true}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Does show first name and last name fields with first name and last name if provided', () => {
		let knownUser = {
			...user,
			first_name: 'Roy',
			last_name: 'Sivan'
		}
		const component = renderer.create(<UserInfo
				user={knownUseru}
				onChangeEmail={genericHandler}
				onUpdateUserFields={genericHandler}
				onLoginAttempt={genericHandler}
				showPassword={true}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});


});




