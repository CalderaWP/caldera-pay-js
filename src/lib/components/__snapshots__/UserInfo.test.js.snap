// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalderaPay components ProductGrid matches snapshot 1`] = `
<table
  className="striped bordered condensed hover table"
>
  <thead>
    <tr>
      <th
        className="sr-only"
      >
        Add-On Name
      </th>
      <th
        className=""
      >
        Free
      </th>
      <th
        className=""
      >
        <a
          href="https://cart.com"
        >
          Bundle 1
        </a>
      </th>
      <th
        className=""
      >
        <a
          href="https://cart.com"
        >
          Bundle 2
        </a>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
      <th>
        <i
          aria-hidden="true"
          className="fa fa-times"
        >
          <span
            className="sr-only"
          >
            Not Included
          </span>
        </i>
      </th>
    </tr>
    <tr>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
      <th
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-check"
        >
          <span
            className="sr-only"
          >
            Hover For More Information
          </span>
        </i>
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`CalderaPay components ProductSearch matches snapshot with a search term 1`] = `
<div>
  <div
    className="caldera-pay-product-search-options"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-product-search"
        >
          Search Products
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-product-search"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value="MailChimp"
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`CalderaPay components ProductSearch matches snapshot with no search term 1`] = `
<div>
  <div
    className="caldera-pay-product-search-options"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-product-search"
        >
          Search Products
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-product-search"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`UserInfo component Does not show password when told not to 1`] = `
<div>
  <div
    className="caldera-pay-customer-information"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-email"
        >
          Email Address
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-email"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-first-name"
        >
          First Name
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-first-name"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-last-name"
        >
          Last Name
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-last-name"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`UserInfo component Does show first name and last name fields with first name and last name if provided 1`] = `
<div>
  <div
    className="caldera-pay-customer-information"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-email"
        >
          Email Address
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-email"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-password"
        >
          Password
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-password"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-first-name"
        >
          First Name
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-first-name"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-last-name"
        >
          Last Name
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-last-name"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`UserInfo component Does show password when told  to 1`] = `
<div>
  <div
    className="caldera-pay-customer-information"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-email"
        >
          Email Address
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-email"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-password"
        >
          Password
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-password"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-first-name"
        >
          First Name
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-first-name"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-customer-last-name"
        >
          Last Name
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-customer-last-name"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
